src/main/java/io/confluent/idesidecar/restapi/connections/CCloudConnectionState.java

```diff
@@ -46,7 +46,7 @@ public class CCloudConnectionState extends ConnectionState {
   }
 
   @Override
-  public ConnectionStatus getInitialStatus() {
+  protected ConnectionStatus getInitialStatus() {
```

Comment:
This is only used at instantiation time and may need to be overridden by subclasses, thus `protected` is more constrained.